src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/GoogleDataprocDAO.scala

```diff
@@ -320,6 +349,19 @@ class GoogleDataprocDAO(protected val dataprocConfig: DataprocConfig, protected
     } yield accessConfig.getNatIP
   }
 
+  private def getInitBucketName(cluster: GoogleCluster): Option[String] = {
+    def parseBucketName(bucketPath: String): Option[String] = {
+      Try(Option(new URI(bucketPath).getHost)).toOption.flatten
+    }
+
+    for {
+      config <- Option(cluster.getConfig)
+      initAction <- config.getInitializationActions.asScala.headOption
+      bucketPath <- Option(initAction.getExecutableFile)
+      bucketName <- parseBucketName(bucketPath)
+    } yield bucketName
+  }
+
   private def executeGoogleRequestAsync[A](googleProject: GoogleProject, clusterName: String, request: AbstractGoogleClientRequest[A])(implicit executionContext: ExecutionContext): Future[A] = {
```

Comment:
Since we're now passing bucketName as well as clusterName, maybe make this more general, like objectName or something?

Reply:
I have had the same thoughts. Similar issue with `CallToGoogleApiFailedException` which takes a `clusterName` argument, but we pass other things like bucketName to it. I'll see if I can make it more general.